external-providers/generic-external-provider/pkg/generic/provider.go

```diff
@@ -37,11 +37,11 @@ func (p *golangProvider) Capabilities() []provider.Capability {
 	}
 }
 
-type golangCondition struct {
+type genericCondition struct {
 	Referenced string `yaml:"referenced"`
```

Comment:
We should actually make `Referenced` a struct here, and give it a field called `Pattern` instead of having `Referenced` be a string itself. This allows us to expand the default implementation of `Referenced` in the future without making breaking changes to the structure of rules (ie, if we added  a generic `Location` field that allowed you to differentiate between declarations and usages).